x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -184,17 +184,15 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
 
   const dispatch = useDispatch();
   const containerElement = useRef<HTMLDivElement | null>(null);
-  const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+  const getTable = useMemo(() => dataTableSelectors.getTableByIdSelector(), []);
   const graphEventId = useShallowEqualSelector(
-    (state) =>
-      (getTimeline(state, TimelineId.detectionsRulesDetailsPage) ?? timelineDefaults).graphEventId
+    (state) => (getTable(state, TableId.detectionsRulesDetailsPage) ?? tableDefaults).graphEventId
   );
   const updatedAt = useShallowEqualSelector(
-    (state) =>
-      (getTimeline(state, TimelineId.detectionsRulesDetailsPage) ?? timelineDefaults).updated
+    (state) => (getTable(state, TableId.detectionsRulesDetailsPage) ?? tableDefaults).updated
   );
   const isAlertsLoading = useShallowEqualSelector(
-    (state) => (getTimeline(state, TimelineId.detectionsPage) ?? timelineDefaults).isLoading
+    (state) => (getTable(state, TableId.detectionsPage) ?? tableDefaults).isLoading
```

Comment:
`TableId.detectionsPage` and `TableId.detectionsRulesDetailsPage` - do you know what tables do these ids correspond to? I believe we have only one table on the Rule Details page that uses TGrid - it's the one in the Alerts tab.

Reply:
@banderror thank you a lot for adding the `Files by Code Owner` (I only with GH has that embedded some day :sparkles:).

> TableId.detectionsPage and TableId.detectionsRulesDetailsPage - do you know what tables do these ids correspond to? I believe we have only one table on the Rule Details page that uses TGrid - it's the one in the Alerts tab.

`TableId.detectionsPage` manage the redux state part of the table on the Alerts page of Security app and `TableId.detectionsRulesDetailsPage` (still alerts :grin:) manage the state part for the Rule Details page. 
This is definitely something to think about - should these two tables have the own state for each... And for now the answer is Yes, because that allows to manage/keep the user settings (columns, sort, pageItems, etc.) per specific table.

Reply:
@YulNaumenko Ah, I see! Thank you :) Yea, I think it makes sense to keep separate user settings for the two tables -- no reason to change that until we get different requirements from the product team.

One suspicious thing here is that `isAlertsLoading` is derived from the state of the table on the Alerts page (`TableId.detectionsPage`) whereas this code executes on the Rule Details page. So I'd expect to see this:

```ts
const isAlertsLoading = useShallowEqualSelector(
  (state) => (getTable(state, TableId.detectionsRulesDetailsPage) ?? tableDefaults).isLoading
```

That looks like a bug and I can debug this further unless you have an immediate answer like "yeah it's a bug" or "no it makes sense because ....".

Reply:
Also, is it possible to give more descriptive names to these table ids or it can be hard because this state is persistent somewhere?

`TableId.detectionsRulesDetailsPage` -> `TableId.alertsOnRuleDetailsPage`
`TableId.detectionsPage` -> `TableId.alertsOnAlertsPage`

or something similar